components/infra-proxy-service/server/orgs.go

```diff
@@ -210,6 +210,44 @@ func (s *Server) ResetOrgAdminKey(ctx context.Context, req *request.ResetOrgAdmi
 	}, nil
 }
 
+//GetInfraServerOrgs: Fetches the list of automate infra server organisations from the chef server and save it into the automate back end DB
+func (s *Server) GetInfraServerOrgs(ctx context.Context, req *request.GetInfraServerOrgs) (*response.GetInfraServerOrgs, error) {
+
+	// Get the credential ID from servers table
+	server, err := s.service.Storage.GetServer(ctx, req.ServerId)
+	if err != nil {
+		return nil, service.ParseStorageError(err, *req, "server")
+	}
+	// Get web ui key from secrets service
+	secret, err := s.service.Secrets.Read(ctx, &secrets.Id{Id: server.CredentialID})
+	if err != nil {
+		return nil, err
+	}
+	// Get organization list from chef server
+	c, err := s.createChefServerClient(ctx, req.ServerId, GetWebuiKeyFrom(secret), "pivotal", true)
+	if err != nil {
+		return nil, err
+	}
+	orgsList, err := c.client.Organizations.List()
+	if err != nil {
+		return nil, ParseAPIError(err)
+	}
+
+	// Save organisations in backend DB
+	orgs := []storage.Org{}
+	for key := range orgsList {
+		org, err := s.service.Storage.StoreOrg(ctx, key, key, "pivotal", "", req.ServerId, nil)
```

Comment:
I guess we can keep the org admin user name as blank which is default

Reply:
Yup, did the changes.